hal/src/nRF52840/pinmap_impl.h

```diff
@@ -15,44 +15,498 @@
  * License along with this library; if not, see <http://www.gnu.org/licenses/>.
  */
 
-#ifndef PINMAP_IMPL_H
-#define PINMAP_IMPL_H
+#pragma once
 
-
-#ifdef  __cplusplus
+#ifdef __cplusplus
 extern "C" {
 #endif
 
 #include "pinmap_hal.h"
 
-typedef struct NRF5x_Pin_Info {
-    uint8_t     gpio_port;      // port0: 0; port: 1;
-    uint8_t     gpio_pin;       // range: 0~31;
-    PinMode     pin_mode;       // GPIO pin mode
-    PinFunction pin_func;       
+typedef struct Hal_Pin_Info {
+    uint8_t     gpio_port; // port0: 0; port: 1;
+    uint8_t     gpio_pin;  // range: 0~31;
+    PinMode     pin_mode;  // GPIO pin mode
+    PinFunction pin_func;
     uint8_t     adc_channel;
     uint8_t     pwm_instance;   // 4 instances on nRF52, range: 0~3
     uint8_t     pwm_channel;    // 4 channels in each instance, range: 0~3
-	uint8_t     pwm_resolution; // default 8bit, max 15bit
+    uint8_t     pwm_resolution; // default 8bit, max 15bit
     uint8_t     exti_channel;   // 16 channels
-    uint32_t    user_data;      
-} NRF5x_Pin_Info;

```

Comment:
Let's leave a compatibility `NRF5x_Pin_Info` typedef here just in case.

Reply:
Added.